dataplane/standalone/saiserver/acl.go

```diff
@@ -0,0 +1,167 @@
+		return nil, status.Errorf(codes.InvalidArgument, "invalid stage type: %v", stage)
+	}
+	if typ != saipb.AclTableGroupType_ACL_TABLE_GROUP_TYPE_PARALLEL {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid type: %v", typ)
+	}
+	if len(bind) != 1 && bind[0] == saipb.AclBindPointType_ACL_BIND_POINT_TYPE_SWITCH {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid bind configuration: %v", typ)
+	}
+
+	tReq := &fwdpb.TableCreateRequest{
+		ContextId: &fwdpb.ContextId{Id: a.dataplane.ID()},
+		Desc: &fwdpb.TableDesc{
+			TableType: fwdpb.TableType_TABLE_TYPE_FLOW,
+			TableId: &fwdpb.TableId{
+				ObjectId: &fwdpb.ObjectId{
+					Id: fmt.Sprint(id),
+				},
+			},
+			Table: &fwdpb.TableDesc_Flow{
+				Flow: &fwdpb.FlowTableDesc{
+					BankCount: 1,
+				},
+			},
+		},
+	}
+	if _, err := a.dataplane.TableCreate(ctx, tReq); err != nil {
+		return nil, err
+	}
+
+	return &saipb.CreateAclTableGroupResponse{Oid: 1}, nil
+}
+
+// CreateAclTableGroupMember stores the acl table id to its corresponding lucius flow table and bank.
+func (a *acl) CreateAclTableGroupMember(_ context.Context, req *saipb.CreateAclTableGroupMemberRequest) (*saipb.CreateAclTableGroupMemberResponse, error) {
+	groupID := req.GetAclTableGroupId()
+	tableID := req.GetAclTableId()
+	memberID := a.mgr.NextID()
+
+	bank := a.groupNextFreeBank[groupID]
+	a.groupNextFreeBank[groupID] = bank + 1
+	a.tableToBank[tableID] = groupBank{
+		groupID:  fmt.Sprint(groupID),
+		bank:     bank,
+		memberID: memberID,
+	}
+
+	return &saipb.CreateAclTableGroupMemberResponse{Oid: memberID}, nil
+}
+
+// CreateAclTable is noop as the table is already created in the group.
+// TODO: Do we need to support tables that aren't in groups.
+func (a *acl) CreateAclTable(_ context.Context, req *saipb.CreateAclTableRequest) (*saipb.CreateAclTableResponse, error) {
+	id := a.mgr.NextID()
+	return &saipb.CreateAclTableResponse{Oid: id}, nil
+}
+
+// CreateAclEntry adds an entry in the a bank.
+func (a *acl) CreateAclEntry(ctx context.Context, req *saipb.CreateAclEntryRequest) (*saipb.CreateAclEntryResponse, error) {
+	id := a.mgr.NextID()
+	gb, ok := a.tableToBank[req.GetTableId()]
+	if !ok {
+		return nil, status.Errorf(codes.FailedPrecondition, "table is not member of a group")
+	}
+
+	aReq := &fwdpb.TableEntryAddRequest{
+		ContextId: &fwdpb.ContextId{Id: a.dataplane.ID()},
+		TableId:   &fwdpb.TableId{ObjectId: &fwdpb.ObjectId{Id: gb.groupID}},
+		EntryDesc: &fwdpb.EntryDesc{
+			Entry: &fwdpb.EntryDesc_Flow{
+				Flow: &fwdpb.FlowEntryDesc{
+					Priority: req.GetPriority(),
+					Id:       uint32(id),
+					Bank:     uint32(gb.bank),
+				},
+			},
+		},
+	}
+	switch {
+	case req.GetFieldDstIp() != nil:
+		aReq.EntryDesc.GetFlow().Fields = append(aReq.EntryDesc.GetFlow().Fields, &fwdpb.PacketFieldMaskedBytes{
+			FieldId: &fwdpb.PacketFieldId{Field: &fwdpb.PacketField{FieldNum: fwdpb.PacketFieldNum_PACKET_FIELD_NUM_IP_ADDR_DST}},
```

Comment:
Just for my understanding, is the IP mask identifier used to identify which flows fall under the table entry? So if the IP mask is 10/8, then 10.10.10.10 would be affected by this table entry?

Reply:
yes